yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -218,11 +218,12 @@ private[spark] class ApplicationMaster(
     }
   }
 
-  private def sparkContextInitialized(sc: SparkContext) = {
+  private def sparkContextInitialized(sc: SparkContext, backend: CoarseGrainedSchedulerBackend) = {
     sparkContextRef.synchronized {
       sparkContextRef.compareAndSet(null, sc)
       sparkContextRef.notifyAll()
     }
+    allocator.setScheduler(backend)
```

Comment:
Hmm, good point - I think there is some merge issue here since it is working for us in cluster mode (hope it is not specific to client case ?).
Let me revisit the PR - apologies if this is a merge mess up !